src/files-and-videos/videos/transcript-settings/OrderTranscriptForm.jsx

```diff
@@ -0,0 +1,165 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage, injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { Button, SelectableBox, Stack } from '@edx/paragon';
+import { ErrorAlert } from '@edx/frontend-lib-content-components';
+import Cielo24Form from './Cielo24Form';
+import ThreePlayMediaForm from './ThreePlayMediaForm';
+import { RequestStatus } from '../../../data/constants';
+import messages from './messages';
+import { checkCredentials, validateForm } from '../data/utils';
+
+const OrderTranscriptForm = ({
+  setTranscriptType,
+  activeTranscriptPreferences,
+  transcriptType,
+  transcriptCredentials,
+  closeTranscriptSettings,
+  handleOrderTranscripts,
+  transcriptionPlans,
+  errorMessages,
+  transcriptStatus,
+  // injected
+  intl,
+}) => {
+  const [data, setData] = useState(activeTranscriptPreferences || { videoSourceLanguage: '' });
+
+  let [cieloHasCredentials, threePlayHasCredentials] = checkCredentials(transcriptCredentials);
+  useEffect(() => {
+    [cieloHasCredentials, threePlayHasCredentials] = checkCredentials(transcriptCredentials);
+  }, [transcriptCredentials]);
+
+  let isFormValid = validateForm(cieloHasCredentials, threePlayHasCredentials, transcriptType, data);
+  useEffect(() => {
+    isFormValid = validateForm(cieloHasCredentials, threePlayHasCredentials, transcriptType, data);
+  }, [data]);
+
+  const handleDiscard = () => {
+    setTranscriptType(activeTranscriptPreferences);
+    closeTranscriptSettings();
+  };
+
+  const handleUpdate = () => handleOrderTranscripts(data, transcriptType);
+
+  let form;
+  switch (transcriptType) {
+  case 'Cielo24':
+    form = (
+      <Cielo24Form
+        {...{
+          hasTranscriptCredentials: cieloHasCredentials,
+          data,
+          setData,
+          transcriptionPlan: transcriptionPlans.Cielo24,
```

Comment:
It's hard to foresee what errors could occur due to misconfiguration of transcriptionPlans and other transcription settings. It would be good to test what happens if `transcriptionPlans` is undefined or `transcriptionPlans.Cielo24` is undefined.